File: server/medicore.apps/src/model/hospitalModel.ts
import { Type } from 'class-transformer';
import { PolyClinicToHospitalMapModel, RegencyModel, IEntityBase, HospitalStaffOperatingHoursModel, HospitalImageModel } from '../model';
import { HospitalStatus } from './enumModel';
import { WriteHistoryBaseModel } from './writeHistoryBaseModel';

export class HospitalModel extends WriteHistoryBaseModel implements IEntityBase {
    id: number;
    name: string;
    address: string;
    zipCode: string;
    latitude?: number;
    longitude?: number;
    regencyId: number;
    isHaveAmbulance: boolean;
    status: HospitalStatus;
    description: string;
    website: string;
    primaryPhone: string;
    secondaryPhone: string;

    //model
    @Type(() => RegencyModel)
    regency: RegencyModel;
    @Type(() => HospitalStaffOperatingHoursModel)
    operatingHours: HospitalStaffOperatingHoursModel[];
    @Type(() => PolyClinicToHospitalMapModel)
    polyClinicMaps: PolyClinicToHospitalMapModel[];
    @Type(() => HospitalImageModel)
    images: HospitalImageModel[];

    constructor() {
        super();
        this.id = 0;
        this.address = null;
        this.zipCode = null;
        this.longitude = null;
        this.latitude = null;
        this.regencyId = 0;
        this.isHaveAmbulance = false;
        this.status = HospitalStatus.Active;
        this.description = null;
        this.website = null;
        this.primaryPhone = null;
        this.secondaryPhone = null;
        this.polyClinicMaps = null;
        this.images = null;
    }
}

File: server/medicore.layer/MediCore.Service/Model/HospitalModel.cs
using dna.core.service.Models.Abstract;
using MediCore.Data.Entities;
using MediCore.Data.Infrastructure;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace MediCore.Service.Model
{
    public class HospitalModel : WriteHistoryBaseModel, IModelBase
    {
        public int Id { get; set; }

        [Required(ErrorMessage = "Required")]
        [StringLength(100, ErrorMessage = "StringLength")]
        public string Name { get; set; }

        [Required(ErrorMessage = "Required")]
        public string Address { get; set; }

        public string ZipCode { get; set; }

        public Nullable<double> Latitude { get; set; }
        public Nullable<double> Longitude { get; set; }
        [Required(ErrorMessage = "Required")]
        public int RegencyId { get; set; }

        [Required(ErrorMessage = "Required")]
        public bool IsHaveAmbulance { get; set; }      
        

        [Required(ErrorMessage = "Required")]
        public HospitalStatus Status { get; set; }

        public string Description { get; set; }

        [StringLength(100, ErrorMessage = "StringLength")]
        public string Website { get; set; }

        [Required(ErrorMessage = "Required")]
        [StringLength(20, ErrorMessage = "StringLength")]
        public string PrimaryPhone { get; set; }

        [StringLength(20, ErrorMessage = "StringLength")]
        public string SecondaryPhone { get; set; }

        [StringLength(100, ErrorMessage = "StringLength")]
        public string PrimaryEmail { get; set; }

        [StringLength(100, ErrorMessage = "StringLength")]
        public string SecondaryEmail { get; set; }

        public double Distance { get; set; }

        // rating summary, it will be fill by AWS Lamda Function
        public double? Rating { get; set; }

        public virtual RegencyModel Regency { get; set; }

        public virtual ICollection<HospitalOperatingHoursModel> OperatingHours { get; set; }
        public virtual ICollection<PolyClinicToHospitalMapModel> PolyClinicMaps { get; set; }
        public virtual ICollection<HospitalImageModel> Images { get; set; }
    }
}
